test/e2e/performanceprofile/functests/utils/client/dataplane.go

```diff
@@ -14,55 +14,55 @@ type dataPlaneImpl struct {
 }
 
 func (dpi *dataPlaneImpl) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
-	if hypershift.IsEncapsulatedInConfigMap(obj) {
+	if hypershift.GetObjectConfigMapDataKey(obj) != "" {
```

Comment:
keep the intent for better readability, reimplement the function:
```
func IsEncapsulatedInConfigMap(obj runtime.Object) bool {
    return GetObjectConfigMapDataKey(obj) != ""
}
```

Reply:
or change again `GetObjectConfigMapDataKey(obj)` so it returns the key and a bool (which feels redundant) and use it like
```
if _, ok := hypershift.GetObjectConfigMapDataKey(obj); !ok {
```
this is again to convey the intent more explicitely

Reply:
I liked the first idea, it's great, thanks